File: app/src/main/java/com/msd/notebook/activity/HomeActivity.java
package com.msd.notebook.activity;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.msd.notebook.R;
import com.msd.notebook.databinding.ActivityHomeBinding;
import com.msd.notebook.fragments.InstructorHomeFragment;
import com.msd.notebook.fragments.QrCodeDisplayFrag;
import com.msd.notebook.fragments.SettingsFragment;

public class HomeActivity extends AppCompatActivity implements BottomNavigationView
        .OnNavigationItemSelectedListener {
    ActivityHomeBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityHomeBinding.inflate(getLayoutInflater());
        View view = binding.getRoot();
        setContentView(view);

        binding.bottomNavigation
                .setOnNavigationItemSelectedListener(HomeActivity.this);
        binding.bottomNavigation.setSelectedItemId(R.id.bottom_navigation);
    }


    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.navigation_qr) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.my_nav_host_fragment, new QrCodeDisplayFrag())
                    .commit();
            return true;
        } else if (item.getItemId() == R.id.navigation_home) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.my_nav_host_fragment, new InstructorHomeFragment())
                    .commit();
            return true;
        } else if (item.getItemId() == R.id.navigation_settings) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.my_nav_host_fragment, new SettingsFragment())
                    .commit();
            return true;
        }
        return false;
    }
}

File: app/src/main/java/com/msd/notebook/activity/HomeActivity.kt
package com.msd.notebook.activity

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.msd.notebook.R
import com.msd.notebook.databinding.ActivityHomeBinding
import com.msd.notebook.fragments.InstructorHomeFragment
import com.msd.notebook.fragments.QrCodeDisplayFrag
import com.msd.notebook.fragments.SettingsFragment

class HomeActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    var binding: ActivityHomeBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHomeBinding.inflate(layoutInflater)
        val view: View = binding!!.getRoot()
        setContentView(view)
        binding!!.bottomNavigation
            .setOnNavigationItemSelectedListener(this@HomeActivity)
        binding!!.bottomNavigation.selectedItemId = R.id.bottom_navigation
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.navigation_qr) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.my_nav_host_fragment, QrCodeDisplayFrag())
                .commit()
            return true
        } else if (item.itemId == R.id.navigation_home) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.my_nav_host_fragment, InstructorHomeFragment())
                .commit()
            return true
        } else if (item.itemId == R.id.navigation_settings) {
            supportFragmentManager
                .beginTransaction()
                .replace(R.id.my_nav_host_fragment, SettingsFragment())
                .commit()
            return true
        }
        return false
    }
}